### PR TITLE
Update (virtually) all test kernels etc. to use operates_on (closes #866)

### LINKED ISSUE #866: [i-first] Change kernel meta-data from "iterates_over" to "operates_on"
For phase I of i-first support for LFRic we need support for a new type of kernel. This kernel (which maps from LFRic to UM Physics kernels) will need to be supplied with entire LFRic fields - as opposed to being called from within a loop over columns or dofs.

### PR DESCRIPTION
**Pull request by arporter**
Only those test kernels required for coverage of the support for the old format are left unchanged.
They will be modified as part of #870.

### CODE REVIEW

**Comment by arporter** [comment]
I have tried to tightly restrict this PR to only changing kernel metadata. Several of the test kernels also had to have their 'use' statements updated in order to bring in 'CELL_COLUMN' instead of 'CELLS'. Ready for review.

**Review by TeranIvy — CHANGES_REQUESTED** [review]
Please update `USE` statements in modified `examples/lfric` kernels 😃

**Comment by arporter** [comment]
Oops, I'd made a typo in my grep. Should be fixed now.

**Review by TeranIvy — CHANGES_REQUESTED** [review]
Thanks for undertaking this giant effort. Thanks also for dilligently referencing issue #870 with the remaining instances of `iterates_over`. Most things look good and the remarks I have are for consistency with the existing code (inline).
- Tests with compilation pass.
- Examples run.
- Modified Python code is `pycodestyle` and `pylint` clean.
- Documentation builds (no change there).
- Coverage is fine and there are examples left for testing of `iterates_over`.

File: examples/lfric/eg2/v1_kernel_mod.F90 [hunk 1]
```diff
@@ -49,7 +49,7 @@ module v1_kernel_mod
 use gaussian_quadrature_mod, only: gaussian_quadrature_type, &
                                    ngp_h, ngp_v ! parameter for how many GQ points
 use argument_mod,            only: arg_type, &          ! the type
-                                   gh_inc, v1, fe, cells ! the enums
+                                   gh_inc, v1, fe, cell_column ! the enums
```
[the remark this answers is not in the document]
Comment (arporter) [reply examples/lfric/eg2/v1_kernel_mod.F90:52]: done

File: examples/lfric/eg2/v2_kernel_mod.F90 [hunk 2]
```diff
@@ -51,7 +51,7 @@ module v2_kernel_mod
 use gaussian_quadrature_mod, only: gaussian_quadrature_type, &
                                    ngp_h,ngp_v
 use argument_mod,            only: arg_type, &          ! the type
-                                   gh_inc, v2, fe, cells ! the enums
+                                   gh_inc, v2, fe, cell_column ! the enums
```
[the remark this answers is not in the document]
Comment (arporter) [reply examples/lfric/eg2/v2_kernel_mod.F90:54]: Done.

**Comment by arporter** [comment]
As it was only a couple of sed's I've made the requested changes although please note we do not mandate any style for the test kernels.
Should be ready for second review assuming Travis is happy.

**Comment by TeranIvy** [comment]
There is no mandated style for Fortran source and examples and there are many source changes. I think it would be good to keep some consistency in these bulk changes. Ideally this would be done on basis of individual examples, but there are many, we have no mandated style and it is ultimately not something critical. 

Looking at the changes and the overall style of the code in modified source, I suggest using the previous upper-case solution for `operates_on` and that is `operates_on = CELL_COLUMN`.  It is a single `grep` command so hopefully it should not be too onerous.

**Comment by arporter** [comment]
Should be ready to review once Travis has reported.

**Review by TeranIvy — APPROVED** [review]
All well now, including the repeated tests with compilation, examples, etc. Proceeding to merge.

**Pull request merged**